src/fprime_gds/plugin/system.py

```diff
@@ -0,0 +1,119 @@
+import inspect
+import logging
+import re
+from typing import Iterable, ForwardRef
+
+import pluggy
+
+
+import fprime_gds.plugin.definitions as definitions
+# For automatic validation of plugins, each plugin class type must be imported here
+import fprime_gds.common.communication.framing as framing
+
+
+PROJECT_NAME = definitions.PROJECT_NAME
+LOGGER = logging.getLogger(__name__)
+
+_NAME_REGEX = re.compile(r"^register_(\w+)_plugin")
+_TYPE_MAPPINGS = {
+    definitions.register_framing_plugin: framing.FramerDeframer
+}
+_SUPPLIED_PLUGIN_MODULES_OR_CLASSES = [
+    framing.FpFramerDeframer
```

Comment:
Nevermind the latest commit clarified the difference

Reply:
One could ask each type for subclasses and then use those.  However, this has one confusing blind-spot:  did that class get imported?  If not, it will not show up.  This is really hard to understand and debug.

Rather, I opted for hand-curated configuration.  That way when you plugin does not show up, a search through that file reveals each step you need to do without some implied magic.

> Remember, this only affects "built-in" plugin selections added by our team.  Any external supplied plugin should publish their work as a python module with a pluggy entrypoint....which will be automatic.